api/v1beta1/cloudstackmachine_types.go

```diff
@@ -85,6 +88,18 @@ type CloudStackResourceIdentifier struct {
 	Name string `json:"name,omitempty"`
 }
 
+type CloudStackResourceDiskOffering struct {
+	CloudStackResourceIdentifier `json:",inline"`
+	// mount point the data disk uses to mount. The actual partition, mkfs and mount are done by cloud-init generated by kubeadmConfig.
+	MountPath string `json:"mountPath"`
+	// device name of data disk
+	Device string `json:"device"`
+	// filesystem used by data disk
+	Filesystem string `json:"filesystem"`
```

Comment:
Where are all these fields used? I don't see them being passed to the CloudStack API anywhere

Reply:
See above comments.